scylla-cql/src/frame/response/result.rs

```diff
@@ -49,9 +49,50 @@ pub struct TableSpec<'a> {
     table_name: Cow<'a, str>,
 }
 
+/// A type of the column in schema metadata or in prepared statement.
+///
+/// In schema metadata this represents simply a column type. In prepared
+/// statement this is used in two places: as a type of bind marker and as a type
+/// of returned value.
+///
+/// Some of the variants contain a `frozen` flag. This flag is only used
+/// in schema metadata. For prepared statement bind markers and result types
+/// those fields will always be set to `false` (even if the DB column
+/// corresponding to given marker / result type is frozen).
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum ColumnType<'frame> {
-    Custom(Cow<'frame, str>),
+    /// Types that are "simple" (non-recursive).
+    Native(NativeType),
+
+    /// Collection types: Map, Set, and List. Those are composite types with
+    /// dynamic size but constant predefined element types.
+    Collection {
+        frozen: bool,
+        typ: CollectionType<'frame>,
+    },
+
+    // A composite list-like type that has a defined size and all its elements
+    // are of the same type. Intuitively, it can be viewed as a list with constant
+    // predefined size, or as a tuple which has all elements of the same type.
+    Vector {
+        typ: Box<ColumnType<'frame>>,
+        dimensions: u16,
+    },
+
+    // A C-struct-like type defined by the user.
+    UserDefinedType {
+        frozen: bool,
+        definition: Arc<UserDefinedType<'frame>>,
+    },
+
+    // A composite type with a defined size and elements of possibly different,
+    // but predefined, types.
+    Tuple(Vec<ColumnType<'frame>>),
```

Comment:
:+1: Very useful docstrings, good job!

:wrench: The lower three variants have `//` instead of `///`.